python/lsst/alert/packet/updateSchema.py

```diff
@@ -0,0 +1,218 @@
+# This file is part of alert_packet.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import fastavro
+import yaml
+import json
+from lsst.alert.packet import get_schema_root, SchemaRegistry
+
+
+__all__ = ['update_schema']
+
+
+def write_schema(schema, version, path):
+
+    updated_schema_loc = path[0:-3] + version[0]+'/'+version[2]
+
+    if not os.path.exists(updated_schema_loc):
+        if not os.path.exists(updated_schema_loc[0:-2]):
+            os.mkdir(updated_schema_loc[0:-2])
+            os.mkdir(updated_schema_loc)
+        else:
+            os.mkdir(updated_schema_loc)
+
+    updated_schema_name = schema['namespace'] + '.' + schema['name'] + '.avsc'
+
+    with open(updated_schema_loc + '/' + updated_schema_name, "w") as f:
+        json.dump(schema, f, indent=2)
+
+
+def add_namespace(schema):
+
+    schema_names = schema['name'].split(".")
+    schema['name'] = schema_names[2]
+    schema['namespace'] = schema_names[0] + "." + schema_names[1]
+    schema = dict(sorted(schema.items(), reverse=True))
+
+    return schema
+
+
+def populate_fields(apdb):
+    """Make a dictionary of fields to populate the avro schema. At present, a
+    number of rules must be included to ensure certain fields are excluded.
+
+    Parameters
+    ----------
+    apdb: `dict`
+        The name of the schema as a string. E.G. diaSource.
+    """
+
+    field_dictionary_array = []
+    for column in apdb['columns']:
+        # We are still finalizing the time series feature names.
+        if (column['name'] != 'validityStart') and (
```

Comment:
I think this would be nicer looking if you made a list of strings to exclude and then used a loop to check rather than retyping `column['name']` a bunch of times.

Reply:
Changed to iterating over a list of excluded fields.